### PR TITLE
Manage templates

### PR DESCRIPTION
**Pull request by louischan-oursky**
connect #167

### CODE REVIEW

File: src/commands/app/updateTemplates.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  return { added, removed, updated, unchanged };
+}
+
+function isEqualReference(a: TemplateItem, b: TemplateItem): boolean {
+  return (
+    a.type === b.type && a.key === b.key && a.language_tag === b.language_tag
+  );
+}
+
+function find<T1 extends TemplateItem, T2 extends TemplateItem>(
```
Comment (kiootic) [inline src/commands/app/updateTemplates.ts]: I think we can use `Array.prototype.find` instead?

File: src/commands/app/updateTemplates.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+): string {
+  const parts = [];
+
+  if (templateItem.key) {
+    parts.push(templateItem.key);
+  }
+
+  parts.push(templateItem.type);
+
+  return join(templateDir, parts.join("/"));
```
Comment (kiootic) [inline src/commands/app/updateTemplates.ts]: We should do `...parts` instead to use OS specific separator.

File: src/container/CLIContainer.ts [hunk 3]
```diff
@@ -270,4 +276,24 @@ export class CLIContainer<T extends BaseAPIClient> extends ControllerContainer<
...
     return resp;
   }
+
+  async uploadTemplate(filePath: string): Promise<string> {
+    const headers: { [name: string]: string } = {};
+    const mediaType = mime.getType(filePath);
+    if (mediaType != null) {
+      headers["content-type"] = mediaType;
+    }
+    const { size } = await fs.stat(filePath);
```
Comment (kiootic) [inline src/container/CLIContainer.ts]: Use `buffer.length` instead would be more accurate.

File: src/commands/app/downloadTemplates.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    }
+  }
+
+  await mkdir(templateDir, { recursive: true });
+
+  for (const i of remoteTemplates) {
+    const p = templateItemToLocalTemplatePath(i, templateDir);
+    const resp = await fetch(i.url);
+    await new Promise((resolve, reject) => {
```
Comment (kiootic) [inline src/commands/app/downloadTemplates.ts]: We should do it as atomic operation; it's better to read entire response before writing it on disk.

**Comment by louischan-oursky** [comment]
@kiootic comments addressed.

**Review by kiootic — APPROVED** [review]

**Review by kiootic — APPROVED** [review]

**Pull request merged**